stingray/io.py

```diff
@@ -653,3 +660,66 @@ def get_btis(gtis, start_time=None, stop_time=None):
 def gti_len(gti):
     """Return the total good time from a list of GTIs."""
     return np.sum([g[1] - g[0] for g in gti])
+
+def _save_pickle_object(object, filename):
+    pickle.dump(object, open(filename, "wb" ))
+
+def _retrieve_pickle_object(filename):
+    return pickle.load(open(filename, "rb" ) )
+
+def _save_hdf5_object(object, filename):
+    pass
+
+def _retrieve_hdf5_object(object, filename):
+    pass
+
+def _save_ascii_object(object, filename):
+    pass
+
+def _retrieve_ascii_object(object, filename):
+    pass
+
+def save_object(object, filename, format):
+    """
+    Pickle a class instance.
+
+    Parameters
+    ----------
+    object: a class instance
```

Comment:
Let's stick to the `pickle` as default. And if the test class is used by all of the 3 test functions (and probably all of the read write functions), and can't be tested if it's a nested class, then it's good to have it outside of the main class.

I have sent a [PR to your branch](https://github.com/usmanwardag/stingray/pull/1). The tests passed [here](https://travis-ci.org/OrkoHunter/stingray/builds/114332577). You can merge it (or cherry-pick the commits).